### ISSUE TITLE
"Craft.appendFootHtml is not a function" JavaScript error in CP in console when certain plugins are installed

### ISSUE THREAD

**Issue by angrybrad** [craftcms/cms#2966]
### Description

It appears that Craft 3.0.10 has some breaking changes for certain plugins.  Confirmed in these, but probably more:

* CP Element Count
* Embedded Assets
* Control Panel CSS
* Rich Variables

Potentially related to https://github.com/craftcms/cms/issues/2965, but not verified.

### Steps to reproduce

1. Install any of these plugins on the latest Craft release.
2. Go to the entry index page and select a checkbox next to an entry.
3. Notice the JavaScript error on the console. Set Status, Actions and New Entry buttons are all broken.

![capture](https://user-images.githubusercontent.com/61869/41070968-761aa698-69aa-11e8-98ef-52f9cc0e0df8.PNG)

### Additional info

- Craft version: 3.0.10.1
- PHP version: 7.1
- Database driver & version: n/a
- Plugins & versions: Latest of any of the above.

**Comment by adampatpattison** [comment]
I am getting this error when trying to select an entry within a Matrix field. 
The index of entries popup shows and then throws the error meaning the only thing that is selectable is the cancel button to close the popup.

 **Additional info**

- Craft version: 3.0.10.3
- PHP version: 7.1.13
- Database driver & version: MySQL 5.5.5

**Comment by brandonkelly** [comment]
@adampatpattison That screencast requires installing Flash Player, which I’d rather not do. Can you please post it somewhere else, or send it to us directly at support@craftcms.com ?

Also, do you have any plugins installed? I haven't been able to reproduce the error locally.

**Comment by nateiler** [comment]
We have a plugin that is doing something similar to entry type switching... an ajax call is fetching fields.  Doing some testing on Craft 3.0.12 and I'm still getting `Uncaught TypeError: Craft.appendFootHtml is not a function` errors when the ajax call response is returned.  Anything we may need to do differently now (trying to find changes on entry type switching to mirror)

**Comment by brandonkelly** [comment]
@nateiler Yeah any custom Ajax requests that don’t go through Craft.postActionRequest() will need to set those headers, to prevent the Ajax request from re-loading JS files / asset bundles that are already included on the current page.

**Comment by adampatpattison** [comment]
Hello Brandon,

2 apologies from me; Firstly, sorry for taking so long to get back to you.
Secondly, sorry for the issues with the video. I just used a screen capture
software and it was their default format, I use VLC player and had no
trouble.

However, it is now inconsequential as I think I have solved/sussed out the
issue. I diagnosed the issue by adding and removing plugins and finally
pinned it down to the Redactor Tweaks plugin from Verbb (
https://github.com/verbb/redactor-tweaks) which was causing the issue!!

I am now upgraded to the latest version of Craft with that removed and my
testing has all passed!

Thanks for your time on this and apologies for any of your time being
wasted.

Keep up the good work!

Thanks,
Pat
